### PR TITLE
Dynamically load helm app into app-store

### LINKED ISSUE #4244: Support import new Helm-based Apps into KubeSphere App Store dynamically
**What's it about?**

The built-in applications are tightly coupled with the KubeSphere version, this is not flexible for us to expand the application ecosystem of KubeSphere App Store and build the partnership with more ISVs and partners. 

I propose to support import new Helm-based apps into KubeSphere App Store dynamically. 

**What's the reason why we need it?**

I believe this is an important feature for Kubesphere. There're a few use cases:

* A lot of ISVs or partners are applied to add their Helm charts to KubeSphere App Store, see an [example](https://github.com/kubesphere/helm-charts/pull/150).
* Users can deploy the applications in time when the related Pull Request has been merged, they don't need to wait for a new release of KubeSphere

Please leave your comments below if there's anyone who agrees with me. Or just give me a thumb up.

**Area Suggestion**

/kind feature-request
/area app-management

### PR DESCRIPTION
**Pull request by xyz-li**
### What type of PR is this?
/kind feature

### What this PR does / why we need it:
Dynamically load helm app into app store.

As our partners need to add their Helm charts to KubeSphere App Store, I add one helm repo type which is called the `built-in` helm repo. The `built-in` repo's name starts with `builtin-` which differentiates it from the repo created by the user.

ks-controller-manager will periodically load apps in the built-in repo and updates the helm repo CRD.  
When the user lists the app in the app store, the ks-apiserver joins the apps in the build-in repo and apps of HelmApplication.

### Which issue(s) this PR fixes:

Fixes #4244

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
Feature: Dynamically load the community developed helm charts into the app store
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/utils/reposcache/repo_cahes.go [hunk 1]
```diff
@@ -118,10 +145,47 @@ func loadBuiltinChartData(name, version string) ([]byte, error) {
...
+	defer c.Unlock()
+
 	c.deleteRepo(repo)
 	return nil
 }
 
+// CopyCategoryCount copies the internal map to avoid `concurrent map iteration and map write`.
+func (c *cachedRepos) CopyCategoryCount() map[string]int {
+	c.RLock()
+	c.RUnlock()
```
Comment (wansir) [inline pkg/utils/reposcache/repo_cahes.go:158]:
```suggestion
	defer c.RUnlock()
```

File: pkg/utils/mathutil/mathutil.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package mathutil
+
+// Max returns the larger of a and b.
+func Max(a, b int) int {
```
Comment (zryfish) [inline pkg/utils/mathutil/mathutil.go:4]: Why not using math.Max https://pkg.go.dev/math#Max

Comment (xyz-li) [reply pkg/utils/mathutil/mathutil.go:4]: The parameters and return value of `math.Max` are all `float64`. `mathutil.Max(a, b)` should written as `int(math.Max(int(a), int(b)))`, there are too many type conversions.

Comment (zryfish) [reply pkg/utils/mathutil/mathutil.go:4]: OK, if you prefer this way, I think it's unnecessary to put it into another package, just make it an inner function, it'll be easy to read.

**Pull request merged**